draft-ietf-webtrans-http3.md

```diff
@@ -568,6 +543,298 @@ A TLS exporter API might permit the context field to be omitted. In this case,
 as with TLS 1.3, the WebTransport Application-Supplied Exporter Context
 becomes zero-length if omitted.
 
+# Flow Control
+
+Flow control governs the amount of resources that can be consumed or data that
+can be sent.  When using WebTransport over HTTP/3, endpoints can limit the
+number of sessions that a peer can create on a single HTTP/3 connection and the
+number of streams that a peer can create within a session.  Endpoints can also
+limit the amount of data that can be consumed by each session and by each
+stream within a session.
+
+WebTransport over HTTP/3 provides connection-level limit that governs the number
+of sessions that can be created on an HTTP/3 connection; see
+{{flow-control-limit-sessions}}.  It also provides the session-level limits that
+govern the number of streams that can be created in a session and limit the
+amount of data that can be exchanged across all streams in each session; see
+{{flow-control-limit-streams}}.
+
+The underlying QUIC connection provides connection and stream level flow
+control.  The connection data limit defines the total amount of data that can be
+sent across all WebTransport sessions and other non-WebTransport streams.  A
+stream's data limit controls the amount of data that can be sent on that
+stream, WebTransport or otherwise; see {{Section 4 of !RFC9000}}.
+
+## Limiting the Number of Simultaneous Sessions {#flow-control-limit-sessions}
+
+This document defines a SETTINGS_WEBTRANSPORT_MAX_SESSIONS parameter that allows
+the server to limit the maximum number of concurrent WebTransport sessions on a
+single HTTP/3 connection.  The client MUST NOT open more sessions than
+indicated in the server SETTINGS parameters.  The server MUST NOT close the
+connection if the client opens sessions exceeding this limit, as the client and
+the server do not have a consistent view of how many sessions are open due to
+the asynchronous nature of the protocol; instead, it MUST reset all of the
+CONNECT streams it is not willing to process with the `HTTP_REQUEST_REJECTED`
+status defined in {{HTTP3}}.
+
+Just like other HTTP requests, WebTransport sessions, and data sent on those
+sessions, are counted against flow control limits.  This document does not
+introduce additional mechanisms for endpoints to limit the relative amount of
+flow control credit consumed by different WebTransport sessions, however
+servers that wish to limit the rate of incoming requests on any particular
+session have alternative mechanisms:
+
+* The `HTTP_REQUEST_REJECTED` error code defined in [HTTP3] indicates to the
+  receiving HTTP/3 stack that the request was not processed in any way.
+* HTTP status code 429 indicates that the request was rejected due to rate
+  limiting {{!RFC6585}}.  Unlike the previous method, this signal is directly
+  propagated to the application.
+
+## Limiting the Number of Streams Within a Session {#flow-control-limit-streams}
```

Comment:
We should probably point out somewhere inside this section that this mechanism only works when you have reliable resets for stream headers.

Reply:
Added some initial text for this in 78d2456